docs/guides/app-payments/build-on-mobile.mdx

```diff
@@ -92,11 +247,81 @@ void openMobilePay(String deepLink) {
     }
 }
 ```
-
 </TabItem>
+</Tabs>
+
+After this, the MobilePay app will be launched and the user will be able to proceed with the payment flow.
+
+## Returning to Your App
+
+There are multiple ways a user can return to your app:
+
+- User approves the payment and is redirected back to your app.
+- User cancels the payment and is redirected back to your app.
+- User fails to complete the payment due to some kind of error and is redirected back to your app.
+- User closes the app before/after reservation using the native app switcher.
+- User presses back in the MobilePay Login screen.
+
+All these can be grouped into:
+
+- App switching.
+- Redirect.
+
+The following section addresses how each one should be handled.
+
+### App Switching
+
+At any point user can close MobilePay app and just return to your app. **This can happen even if payment is already reserved!** Since it is unknown if your app was opened after completed payment, you'll need to check payment state each time your app returns to the foreground. For this, you should use your backend service that provides current state of payment (more details [here](/docs/payments-refunds/create-payments)).
```

Comment:
```suggestion
At any point the user can close the MobilePay app and just return to your app. **This can happen even if the payment is already reserved!** Since it is unknown if your app was opened after a completed payment, you'll need to check the payment state each time your app returns to the foreground. For this, you should use your backend service that provides the current state of a payment (more details [here](/docs/payments-refunds/create-payments)).
```